features/pakiti/client/config.pan

```diff
@@ -6,34 +6,87 @@ include PAKITI_RPMS;
+variable PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE ?= {
+    if ( file_exists(PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT) ) {
+        PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT;
+    } else {
+        undef;
+    };
+};
 
+# Build configuration file
+include 'components/metaconfig/config';
+include 'features/pakiti/client/config_schema';
+prefix '/software/components/metaconfig/services/{/etc/pakiti/pakiti-client.conf}';
+'backup' = '.old';
+'module' = 'tiny';
+'contents' = {
+    SELF['site'] = PAKITI_TAG;
+    SELF['url'] = format(
+        "%s://%s:%s%s",
+        PAKITI_SERVER_PROTOCOL,
+        PAKITI_SERVER,
+        PAKITI_SERVER_HTTP_PORT,
+        PAKITI_SERVER_HTTP_FEED_URL,
+    );
+    if ( is_defined(PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE) ) {
+        SELF['curl'] = format("curl --cacert %s", PAKITI_SERVER_CERTIFICATION_CHAIN_FILE);
+    };
+    # If PAKITI_SERVER_PUB_KEY_SOURCE is an existing file, define the encrypt
+    # parameter. Else raise an error if the protocol is not https.
+    if ( file_exists(PAKITI_SERVER_PUB_KEY_SOURCE) ) {
+        SELF['encrypt'] = PAKITI_SERVER_PUB_KEY_FILE;
+    } else if ( PAKITI_SERVER_PROTOCOL != 'https' ) {
+        error('Pakiti server pub key is missing and feed protocol is not https');
+    };
+    SELF['rndsleep'] = PAKITI_CLIENT_SLEEP;
+    SELF;
+};
+'convert/yesno' = true;
+bind '/software/components/metaconfig/services/{/etc/pakiti/pakiti-client.conf}/contents' = pakiti_client_config;
 
+# Generate server certificate and certification chain
 include "components/filecopy/config";
-"/software/components/filecopy/services" = npush(
-    escape(PAKITI_CLIENT_CONF), dict(
-        "config", pakiti_conf,
-        "perms", "0644",
-    ),
-);
+"/software/components/filecopy/services" = {
+    debug('Pakiti server certification chain source = %s', to_string(PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE));
+    if ( is_defined(PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE) ) {
+        SELF[escape(PAKITI_SERVER_CERTIFICATION_CHAIN_FILE)] =  dict(
+            "config", file_contents(PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE),
+            "perms", "0644",
+        );
+    };
+    if ( file_exists(PAKITI_SERVER_PUB_KEY_SOURCE) ) {
+        SELF[escape(PAKITI_SERVER_PUB_KEY_FILE)] = dict(
+            "config", file_contents(PAKITI_SERVER_PUB_KEY_SOURCE),
+            "perms", "0644",
+        );
+    };
+    SELF;
+};
 
+# Add cron to run Pakiti client
 include "components/cron/config";
-"/software/components/cron/entries" =
-    push(dict("name", "pakiti_update",
-            "user", "root",
-            "frequency", PAKITI_CLIENT_UPDATE_FREQ,
-            'command', '/usr/bin/pakiti-client --conf ' + PAKITI_CLIENT_CONF));
+"/software/components/cron/entries" = {
+    SELF[length(SELF)] = dict(
+        "name", "pakiti_update",
+        "user", "root",
+        "frequency", PAKITI_CLIENT_UPDATE_FREQ,
+        "command", "/usr/bin/pakiti-client --conf /etc/pakiti/pakiti-client.conf",
```

Comment:
If this is for Pakiti v3 can we make this:
```suggestion
        "command", "/usr/bin/pakiti-client --conf /etc/pakiti3/pakiti-client.conf",
```

Reply:
Done